File: parralel-prog/Lab3/F2.cs
using System;
using System.Threading;
using static lab3.Lab3;

namespace lab3
{
    public class Function2
    {
        public void Run()
        {
            // F2 2.2   MF = MG*(MK*ML) - MK 
             lock (Lab3.locker)
             {
                
                //Thread.Sleep(1000);
                Matrix MG = new Matrix(N);
                Matrix MK = new Matrix(N);
                Matrix ML = new Matrix(N);
                
                Matrix result;

                MG.FillMatrixWithNumber(1);
                MK.FillMatrixWithNumber(1);
                ML.FillMatrixWithNumber(1);

                Console.WriteLine("Function 2 start");
                result = MG.multiply(MK.multiply(ML)).sub(MK);

                Console.WriteLine("Function 2 result \n" + result.toString());
                Console.WriteLine("Function 2 end \n");
             }
        }
    }

}

File: parralel-prog/Lab5/F2.cpp
#include "F2.h"

F2::F2(int N) {
	this->N = N;
}

Matrix* F2::getResult() {
	return result;
}

DWORD WINAPI F2::startThread(void* param) {
	F2* This = (F2*)param;
	return This->run();
}

DWORD F2::run() {
	cout << "Task 2 start\n";
	Matrix *ML = new Matrix(N);
	Matrix *MH = new Matrix(N);
	Matrix *MG = new Matrix(N);
	result = ML->sum(MH->multiply(MG));
	std::cout << result->toString() << std::endl;
	cout << "Task 2 end\n";
	delete ML;
	delete MG;
	delete MH;
	return 0;
}